### PR TITLE
[Merged by Bors] - feat: the homotopy category of cochain complexes is pretriangulated

### PR DESCRIPTION
**Pull request by joelriou**
This PR defines the pretriangulated structure on the homotopy category of cochain complexes in an additive category.

This result first appeared in the Liquid Tensor Experiment. In the LTE, the formalization followed the Stacks Project: in particular, the distinguished triangles were defined using degreewise-split short exact sequences of cochain complexes. Here, we follow the original definitions in Verdier's thesis (with the better sign conventions from the introduction of Brian Conrad's book *Grothendieck duality and base change*).

---
- [x] depends on: #8966
- [x] depends on: #8969
- [x] depends on: #8937 
- [x] depends on: #9054  
- [x] depends on: #8951
- [x] depends on: #9049 
- [x] depends on: #9073 
- [x] depends on: #9592 
- [x] depends on: #9614
- [x] depends on: #9615

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

File: Mathlib/Algebra/Homology/HomotopyCategory/Pretriangulated.lean [hunk 1]
```diff
@@ -9,8 +9,13 @@ import Mathlib.CategoryTheory.Triangulated.Functor
 
 /-! The pretriangulated structure on the homotopy category of complexes
 
+<<<<<<< HEAD
```
Comment (jcommelin) [inline Mathlib/Algebra/Homology/HomotopyCategory/Pretriangulated.lean:12]: Seems like something went wrong with merge conflict resolution.

Comment (joelriou) [reply Mathlib/Algebra/Homology/HomotopyCategory/Pretriangulated.lean]: Oops. I have just fixed this.

File: Mathlib/Algebra/Homology/HomotopyCategory/Pretriangulated.lean [hunk 2]
```diff
@@ -359,3 +359,110 @@ end Shift
...
 
 end CochainComplex
+
+namespace HomotopyCategory
+
+variable (C)
+
+namespace Pretriangulated
+
+/-- A triangle in `HomotopyCategory C (ComplexShape.up ℤ)` if it is isomorphic to
```
Comment (dagurtomas) [inline Mathlib/Algebra/Homology/HomotopyCategory/Pretriangulated.lean:369]:
```suggestion
/-- A triangle in `HomotopyCategory C (ComplexShape.up ℤ)` is distinguished if it is isomorphic to
```

**Review by dagurtomas — APPROVED** [review]
I think this looks great!

**Review by jcommelin — APPROVED** [review]
bors d+

**Comment by joelriou** [comment]
Thanks @dagurtomas and @jcommelin for the review!

bors merge

**Pull request closed without merging**